docs/src/DeveloperGuide/ArchitectureOverview.md

```diff
@@ -118,4 +118,101 @@ needs while benefiting from the components that QAT ships with.
+                                              │                   │
+                                              ▼                   ▼
+                                     ┌ ─ ─ ─ ─ ─ ─ ─ ─ ┐┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┐
+                                       Standard error     Standard Output:
+                                     │    or file:     ││   Resulting IR    │
+                                          JSON Logs
+                                     └ ─ ─ ─ ─ ─ ─ ─ ─ ┘└ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘
+```
+
+The profile consists of a generator and a validator. The generator is
+responsible for performing as many transformations as possible to get the
+original QIR to be compliant with the selected profile. This is done by each of
+the components taking a configuration which then installs LLVM passes to execute
+said transactions. This is illustrated on the left hand-side in the following
+figure whereas the pass execution is illustrated on the right hand-side:
+
+```text
+                                                    ┌ ─ ─ ─ ─ ─ ─ ─ ┐
+                                                       LLVM module
+┌───┐                                               └ ─ ─ ─│─ ─ ─ ─ ┘
+│   │                                                      │ Apply profile
+│ C │                                                      │ process
+│ o │                                     ┌ Generator ─ ─ ─│─ ─ ─ ─ ─
+│ n │If                                            ┌───────▼───────┐ │
+│ f │component                     Install│        │  Inline pass  │
+│ i │active:  ┌───────────────────┐passes   ┌───┐  └───────┬───────┘ │
+│ g ├─────────▶    LLVM passes    │───────┼─▶   │  ┌───────▼───────┐
+│ u │         └───────────────────┘         │   │  │  Unroll pass  │ │
+│ r │                                     │ │ P │  └───────┬───────┘
+│ a │         ┌───────────────────┐         │ a │  ┌───────▼───────┐ │
+│ t ├─────────▶  Transformation   │───────┼─▶ s │  │Transform pass │
+│ i │         └───────────────────┘         │ s │  └───────┬───────┘ │
+│ o │                                     │ │   │  ┌───────▼───────┐
+│ n │         ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┐         │ m │  │  Inline pass  │ │
+│   ├─────────▶    LLVM passes     ───────┼─▶ a │  └───────┬───────┘
+│ m │         └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘         │ n │  ┌───────▼───────┐ │
+│ a │                                     │ │ a │  │   Fold pass   │
+│ n │         ┌───────────────────┐         │ g │  └───────┬───────┘ │
+│ a ├─────────▶     Grouping      │───────┼─▶ e │  ┌───────▼───────┐
+│ g │         └───────────────────┘         │ r │  │  Group pass   │ │
+│ e │                                     │ │ s │  └───────┬───────┘
+│ r │                                       │   │          │         │
+│   │                                     │ │   │          │
+└───┘                                       └───┘          │         │
+                                          └ ─ ─ ─ ─ ─ ─ ─ ─│─ ─ ─ ─ ─
```

Comment:
I see the right part somewhat misaligned.
Pass managers, Unroll pass, Transform pass, Inline pass, Group pass
Maybe that's just a font of the arrow-head.

Reply:
Hmmm... that might be a font issue. It looks OK on GH and in Sublime Text on my machine - could you check with you are using a monospace font?